### ISSUE TITLE
2.1.2: No keyboard trap. Is it OK if a component can only be escaped by back-tabbing?

### ISSUE THREAD

**Issue by Jym77** [w3c/wcag#2678]
2.1.2 states " if [moving focus] requires more than unmodified arrow or tab keys or **other standard exit methods**, the user is advised of the method for moving focus away." (emphasise mine)

The question is what is accepted by "other standard exit methods"?

Specifically, if there is a component which traps focus when forward tabbing (using TAB only), but let it go with backward tabbing (using SHIFT + TAB), is this a failure? (possibly, the full tab-order can be cycled through by use of backward tabbing, but forward tabbing hits a trap).

If there is a component that can only be exited with forward tabbing, this is covered by the precision above. Moving focus can be done by unmodified tab key, which satisfies 2.1.2.

But backward tabbing is not unmodified tab key… So, is it considered "standard exit method" and satisfies 2.1.2? Or does it needs specific instructions as per the rest of sentence?

We've encountered the question in ACT rules, see https://github.com/act-rules/act-rules.github.io/issues/1885 for a bit more discussion.

**Comment by patrickhlauke** [comment]
For what it's worth, I've hit this question myself a few times. In the end, opted to fail it under 2.4.3 Focus Order instead (while mentioning 2.1.2 Keyboard Trap in a "it may pass the wording, so nominally a pass, but it's still broken")

**Comment by fstrr** [comment]
This issue looks like it can be closed. If it needs to be re-opened, please start a new discussion.
